Umee-Bots/lendingPool-reentrancy/src/agent.ts

```diff
@@ -0,0 +1,38 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+
+import utils from "./utils";
+
+export const handleTransaction =
+  (contractAddress: string, reenterancyFunctionsSignatures: string[]): HandleTransaction =>
+  async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const { traces } = txEvent;
+    for (let i = 0; i < traces.length; i++) {
+      const trace = traces[i];
+      const depth = trace.traceAddress.length;
+
+      if (trace.action.to === contractAddress) {
+        let j;
+
+        for (j = i + 1; j < traces.length; j++) {
+          if (traces[j].traceAddress.length === depth) {
```

Comment:
Actually, in this case it'd be `traces[j].traceAddress.length <= depth`, right?
Example: `[[], [0], [0, 0], [0, 0, 0], [1]]`. Let's say `[0, 0, 0]` is a call to `LendingPool`, then the subtree ends in `[1]`.